services/src/main/java/org/fao/geonet/api/records/MetadataInsertDeleteApi.java

```diff
@@ -215,6 +216,9 @@ public void deleteRecord(
             approved=false;
         }
 
+        MetadataPreRemove preRemoveEvent = new MetadataPreRemove(metadata);
```

Comment:
If this fails then there will still be a backup record created.
I think this needs to be moved up before the backup as there is no need to make a backup if nothing was deleted.

Reply:
Moved